data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,675 @@
+- : int list = [0; 1; 1; 2; 3; 5; 8; 13; 21; 34]
+```
+
+The first version (`fibo`) takes two extra parameters: the two previously computed Fibonacci numbers.
+
+The second version (`fib`) uses the first two Fibonacci numbers as initial values. There is nothing to be computed when returning from a recursive call, so this enables the compiler to perform an optimisation called “tail call elimination.” This turns recursivity into imperative iteration in the generated native code and leads to improved performances.
+
+> Note: Notice that the `fib` function has three parameters (`m n i`) but only two arguments were passed (`0 1`). This is called *partial application*.
+
+## Multiple Arguments Functions
+
+### With Syntactic Sugar
+
+To define a function with multiple arguments, each must be listed between the name of the function (right after the `let` keyword) and the equal sign, separated by space. Here is an example:
+```ocaml
+# let sweet_cat x y = x ^ " " ^ y;;
+val sweet_cat : string -> string -> string = <fun>
+
+# sweet_cat "kitty" "cat";;
+- : string = "kitty cat"
+```
+
+This is how most multiple-argument functions are defined. Alternatives exist, but this should be the default.
+
+### Without Syntactic Sugar
+
+The function `sweet_cat` is the same as this one:
+```ocaml
+# let sour_cat = fun x -> fun y -> x ^ " " ^ y;;
+val sour_cat : string -> string -> string = <fun>
+
+# sweet_cat "kitty" "cat";;
+- : string = "kitty cat"
+```
+
+Observe `sweet_cat` and `sour_cat` have the same body: `x ^ " " ^ y`. They only differ in the way parameters are listed:
+1. As `x y` between name and `=` in `sweet_cat`
+2. As `fun x -> fun y ->` after `=` in `sour_cat` (and nothing but name before `=`)
+
+Also observe that `sweet_cat` and `sour_cat` have the same type: `string -> string -> string`.
+
+In reality, these two definitions are the same. If you check the assembly code generated using [compiler explorer](https://godbolt.org/), you'll see it is the same.
+
+The way `sour_cat` is written corresponds more explicitly to the behaviour of both functions. The name `sour_cat` is bound to an anonymous function having parameter `x` and returning an anonymous function having parameter `y` and returning `x ^ " " ^ y`.
+
+The way `sweet_cat` is written is an abbreviated version of `sour_cat`. Such a way of shortening syntax is called [syntactic sugar](https://en.wikipedia.org/wiki/Syntactic_sugar).
+
+### Partial Application and Closures
+
+Having multiple-argument functions as a series of nested single-argument functions is what makes partial application possible.
+```ocaml
+# let sour_kitty x = sour_cat "kitty" x;;
+val sour_kitty : string -> string = <fun>
+
+# let sweet_kitty = fun x -> sweet_cat "kitty kitty" x;;
+val sweet_kitty : string -> string = <fun>
+
+# sour_kitty "cat";;
+- : string = "kitty cat"
+
+# sweet_kitty "cat";;
+- : string = "kitty kitty cat"
+```
+
+Both `sweet_cat` and `sour_cat` allow partial application.
```

Comment:
```suggestion
Both `sweet_cat` and `sour_cat` allow partial application.
```

If there's only one parameter `x` after the `=` (like in `let sour_kitty x = sour_cat "kitty" x;;`), how can it be partially applied?